lib/util/glob-util.js

```diff
@@ -160,10 +154,25 @@ function listFilesToProcess(globPatterns, options) {
         const file = path.resolve(cwd, pattern);
 
         if (shell.test("-f", file)) {
-            addFile(fs.realpathSync(file), !shell.test("-d", file));
+            const ignoredPaths = new IgnoredPaths(options);
+
+            addFile(fs.realpathSync(file), !shell.test("-d", file), ignoredPaths);
         } else {
+
+            // regex to find .hidden or /.hidden patterns, but not ./relative or ../relative
+            const globIncludesDotfiles = /(?:(?:^\.)|(?:[\/\\]\.))[^\/\\\.].*/.test(pattern);
+
+            options.dotfiles = options.dotfiles || globIncludesDotfiles;
```

Comment:
Hmm... We seemed to leaked a bunch of undocumented features into our options:-) Ok, another question then, would this start linting dotfiles automatically, or does user needs to do something to include them?